src/libs/Violations/ViolationsUtils.ts

```diff
@@ -24,15 +24,16 @@ const ViolationsUtils = {
         if (policyRequiresCategories) {
             const hasCategoryOutOfPolicyViolation = transactionViolations.some((violation) => violation.name === 'categoryOutOfPolicy');
             const hasMissingCategoryViolation = transactionViolations.some((violation) => violation.name === 'missingCategory');
-            const isCategoryInPolicy = !!policyCategories[transaction.category ?? '']?.enabled;
+            const categoryKey = updatedTransaction.category;
+            const isCategoryInPolicy = categoryKey ? policyCategories?.[categoryKey]?.enabled : undefined;
```

Comment:
Can/should we fallback to `false` here instead of `undefined`? I think false is a better value for `isCategoryInPolicy`
```suggestion
            const isCategoryInPolicy = categoryKey ? policyCategories?.[categoryKey]?.enabled : false;
```

Reply:
Sure, that's fine.  I think it can be a fine semantic distinction between "not in policy' and "policy category missing", but this will work too.